README-cmake-details.md

```diff
@@ -0,0 +1,223 @@
+# Integrating a Rust Component into an ESP-IDF Project
+
+ESP-IDF, the official development framework for the ESP32 Series SoCs, supports integration of components written in C/C++ and Rust which is gaining traction for embedded development due to its safety features. This article outlines the project layout generated by using the `esp-idf-template` utility and its support for using Rust in CMake-based ESP-IDF projects.
+
+Note that this article assumes prior knolwedge in the ESP-IDF build system (CMake).
+
+## Prerequisites
+
+Follow all the steps in the [CMake tutorial](../README-cmake.md). With the steps outlined there, generate a project named `test`. I.e. 
+
+```sh
+cargo generate --vcs none --git https://github.com/esp-rs/esp-idf-template cmake --name test
+```
+
+## Generated Project Structure
+
+Here's how your project directory might look after following the guide:
+
+```
+test/
+|-- CMakeLists.txt
+|-- main/
+|   |-- CMakeLists.txt
+|   |-- main.c
+|-- sdkconfig
+|-- components/
+|   |-- rust-test/
+|       |-- CMakeLists.txt
+|       |-- placeholder.c
+|       |-- build.rs
+|       |-- Cargo.toml
+|       |-- rust-toolchain.toml
+|       |-- src/
+|           |-- lib.rs
+```
+
+### Key Elements
+
+- `test` contains the main C code like any other ESP-IDF application.
+- An ESP-IDF component, with name `rust-test` is stored in a subdirectory named `components`.
+- The Rust code is stored directly as part of the `rust-test` ESP-IDF component, i.e. the `rust-test` ESP-IDF component is simultaneously an ESP-IDF component as well as a Rust library crate.
+
+The component can be uploaded later to [Component Manager](https://components.espressif.com/).
+
+## How the Build Process Works
+
+The CMake build for your `rust-test` component would proceed as usual and will compile all C code in the component (which is not much - just an empty `placeholder.c` file which needs to be there for the approach to work).
+
+More importantly, `components/rust-test/CMakeLists.txt` is setup by the generator in such a way, so as to **also** call the Rust standard build utility - [`cargo`](https://doc.rust-lang.org/cargo/index.html) - which would take care of compiling the Rust source code in your ESP-IDF component.
+
+The Rust source code (or as it is named in the Rust world - the Rust *crate*) is setup - in `components/rust-test/Cargo.toml` - to have the shape of a *static* library - just like any other ESP-IDF component. When the CMake build calls `cargo` to compile your `rust-test` Rust crate, you'll essentially end up with a static library named `librust_test.a`, which is then linked to the rest of the ESP-IDF project by CMake.
+
+## Project Structure Details
```

Comment:
Maybe place this section under "Generated Project Structure" or at least, after it.

Reply:
Moved the "How The build Works" as a subsection of "Project Structure".